src/lib/esnext.object.d.ts

```diff
@@ -0,0 +1,13 @@
+interface ObjectConstructor {
+    /**
+     * Returns an object from an array of properties key/values
+     * @param entries Array that contains the properties key and value.
+     */
+    fromEntries<T>(entries: ArrayLike<[string, T]> | Iterable<[string, T]>): Record<string, T>;
```

Comment:
Drive-by comment: `fromEntries` allows Symbol keys, not just strings.

Reply:
![image](https://user-images.githubusercontent.com/73085/56263302-605a7e80-611d-11e9-89b4-17197e1341dd.png)

😸

The more comprehensive version is:

1) The first argument **must** be an `Iterable`. The algorithm literally calls `@@iterator`. Can't get away with `ArrayLike`s here.
2) The iterable must yield an object of the shape `{ 0?: string | number | symbol | null | { valueOf(): string | number } | { toString(): string }; 1?: any }`. This is probably not very useful for the actual type definition, and is incomplete as the `valueOf` may just be any other coercible value; but the point is that the `Iterable`'s value needs not be an array or `ArrayLike`.
2a) The `"0"` key / first element of tuple just needs to be something that can be cast into an object key, and that includes `undefined`. `undefined` becomes `"undefined"` and `null` becomes `"null"`. Pretty much the only thing I found that "doesn't work" is `Object.create(null)`.
2b) The `"1"` key / second element of tuple is the value associated with the key.
2c) Any of those keys may be getters or just missing entirely. They may even be write-only setters.
3) Duplicates overwrite each other in iteration order. The last value wins.

---

Current TypeScript treats tuples as assignable to objects with numeric literal keys. Using `<K extends keyof any, T>` as the generic argument and `Iterable<{ 0: K, 1: T }>` _should_ work for most cases; with maybe a few more overloads for index signatures and a catch-all for mixed tuples.